### PR TITLE
fix: Add {{workflow.status}} to workflow-metrics

### PR DESCRIPTION
**Pull request by simster7**

### CODE REVIEW

File: workflow/controller/operator.go [hunk 1]
```diff
@@ -399,6 +399,7 @@ func (woc *wfOperationCtx) operate() {
 		realTimeScope := map[string]func() float64{common.GlobalVarWorkflowDuration: func() float64 {
 			return node.FinishedAt.Sub(node.StartedAt.Time).Seconds()
 		}}
+		woc.globalParams[common.GlobalVarWorkflowStatus] = string(workflowStatus)
```
Comment (alexec) [inline workflow/controller/operator.go:402]: test?

Comment (simster7) [reply workflow/controller/operator.go:402]: added

File: workflow/controller/operator_test.go [hunk 2]
```diff
@@ -3739,3 +3739,113 @@ func TestValidReferenceMode(t *testing.T) {
 	woc.operate()
 	assert.Equal(t, wfv1.NodeRunning, woc.wf.Status.Phase)
 }
+
+var workflowStatusMetric = `
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
```
Comment (alexec) [inline workflow/controller/operator_test.go:3745]: anyway to tidy this to just the relevant fields? hard to understand as it is

Comment (simster7) [reply workflow/controller/operator_test.go:3745]: This is as tidy as possible. This tests relies on using the raw K8s object as a snapshot of its state right before the behavior we are interested in testing.

Most of our operator tests use this pattern.

**Pull request merged**